src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -80,18 +81,80 @@ class ActivityFeed extends React.Component<Props, State> {
     approvalCommentFormSubmitHandler = (): void => this.setState({ isInputOpen: false });
 
     /**
-     * Creates a comment
+     * Add a placeholder pending feed item.
      *
-     * @param {string} text - Comment text
-     * @param {boolean} hasMention - If this comment contains an at mention
+     * @param {Object} itemBase - Base properties for item to be added to the feed as pending.
+     * @return {void}
+     */
+    addPendingItem = (itemBase: Object): void => {
+        const { currentUser } = this.props;
+        // create a placeholder pending feed item
+        const date = new Date().toISOString();
+        const feedItem = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...itemBase
+        };
+
+        const { feedItems } = this.state;
+        feedItems.unshift(feedItem);
+        this.setState({ feedItems });
+    };
+
+    /**
+     * Replace a pending feed item with actual feed item data.
+     *
+     * @param {Comment | Task} feedItem - API returned feed item data.
+     * @param {string} uuid - Unique ID of the pending item to replace.
+     * @return {void}
+     */
+    updatePendingItem = (feedItem: Comment | Task, uuid: string): void => {
+        let itemIndex = null;
+        const { feedItems } = this.state;
+        feedItems.find((item, index) => {
+            if (item.id === uuid) {
+                itemIndex = index;
+                return true;
+            }
+            return false;
+        });
+
+        // Replace item in the feed items or set as most recent item.
```

Comment:
Can you explain this case a bit more? If you're updating a pending item, then why would you be straight up adding it to the sorted array? Shouldn't there always be a pending item to be replaced?

Reply:
In this case, there really shouldn't be a missing pending item. If there is, we just won't do anything. Updated that code